Sources/OpenAPIRuntime/Base/CopyOnWriteBox.swift

```diff
@@ -0,0 +1,197 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A type that wraps a value and enforces copy-on-write semantics.
+///
+/// It also enables recursive types by introducing a "box" into the cycle, which
+/// allows the owning type to have a finite size.
+@_spi(Generated)
+public struct CopyOnWriteBox<Wrapped> {
+
+    /// The reference type storage for the box.
+    @usableFromInline
+    internal final class Storage {
+
+        /// The stored value.
+        @usableFromInline
+        var value: Wrapped
+
+        /// Creates a new storage with the provided initial value.
+        /// - Parameter value: The initial value to store in the box.
+        @usableFromInline
+        init(value: Wrapped) {
+            self.value = value
+        }
```

Comment:
`Storage` is implicitly generic because its outer type is generic so this should be `@inlinable` as well.

Reply:
https://github.com/apple/swift-openapi-runtime/pull/58/commits/9f1e5cdd83c2e434c8c33b8d6dbf9895f19a7f70